File: apps/merchant-api/InventoryService/src/InventoryService.Domain/Concretes/ProductReview.cs
using InventoryService.Domain.Bases;

namespace InventoryService.Domain.Concretes;

public class ProductReview : BaseEntity
{
    public Guid UserId { get; set; }
    public Guid ProductId { get; set; }
    public int Rating { get; set; }
    public string Comment { get; set; } = string.Empty;
    public Product? Product { get; set; }
}

File: apps/merchant-web/src/commons/entities/concretes/ProductReview.ts
import { UUID } from "crypto";
import Product from "./Product";

export default class ProductReview {
  id: UUID;
  userId: UUID;
  productId: UUID;
  rating: number;
  comment: string;
  product: Product | null;

  constructor(id: UUID, userId: UUID, productId: UUID, rating: number, comment: string, product: Product) {
    this.id = id;
    this.userId = userId;
    this.productId = productId;
    this.rating = rating;
    this.comment = comment;
    this.product = product;
  }
}